Marlin/Marlin_main.cpp

```diff
@@ -9483,25 +9483,39 @@ inline void gcode_M999() {
 }
 
 #if ENABLED(SWITCHING_EXTRUDER)
-  inline void move_extruder_servo(uint8_t e) {
-    const int angles[2] = SWITCHING_EXTRUDER_SERVO_ANGLES;
-    MOVE_SERVO(SWITCHING_EXTRUDER_SERVO_NR, angles[e]);
-    safe_delay(500);
+  inline void move_extruder_servo(const uint8_t e) {
+    stepper.synchronize();
+    constexpr int16_t angles[] = SWITCHING_EXTRUDER_SERVO_ANGLES;
+    static_assert(COUNT(angles) / 2 >= E_STEPPERS, "SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.");
+
+    #if EXTRUDERS & 1
```

Comment:
I'm pretty sure `E_STEPPERS` will always be equal to `EXTRUDERS` with this type of setup, and this feature relates to the steppers, not the nozzles. `EXTRUDERS` should only very rarely be used, since it represents neither nozzles nor steppers.